jenkins/run-build-packages.sh

```diff
@@ -354,6 +355,73 @@ LIBCLOUD_DIR=$(mktemp -d)
 fpm_build $LIBCLOUD_DIR "$PYTHON2_PKG_PREFIX"-apache-libcloud
 rm -rf $LIBCLOUD_DIR
 
+if [[ $TARGET =~ ubuntu1204 ]]; then
+    # port libfuse 2.9.2 to Ubuntu 12.04 
+    LIBFUSE_DIR=$(mktemp -d)
+    (
+        cd $LIBFUSE_DIR
+        # download fuse 2.9.2 ubuntu 14.04 source package
+        curl -o fuse_2.9.2.orig.tar.xz http://archive.ubuntu.com/ubuntu/pool/main/f/fuse/fuse_2.9.2.orig.tar.xz
+        curl -o fuse_2.9.2-4ubuntu4.14.04.1.debian.tar.xz http://archive.ubuntu.com/ubuntu/pool/main/f/fuse/fuse_2.9.2-4ubuntu4.14.04.1.debian.tar.xz
+        curl -o fuse_2.9.2-4ubuntu4.14.04.1.dsc http://archive.ubuntu.com/ubuntu/pool/main/f/fuse/fuse_2.9.2-4ubuntu4.14.04.1.dsc
+
+        # install dpkg-source and dpkg-buildpackage commands
+        apt-get install -y dpkg-dev
+
+        # extract source and apply patches
+        dpkg-source -x fuse_2.9.2-4ubuntu4.14.04.1.dsc
+        rm -f fuse_2.9.2.orig.tar.xz fuse_2.9.2-4ubuntu4.14.04.1.debian.tar.xz fuse_2.9.2-4ubuntu4.14.04.1.dsc
+
+        # add new version to changelog
+        cd fuse-2.9.2
+	mv debian/changelog debian/changelog.old
+	echo "fuse (2.9.2-5) precise; urgency=low" > debian/changelog
+	echo "" >> debian/changelog
+	echo "  * Backported from trusty-security to precise" >> debian/changelog
+	echo "" >> debian/changelog
+	echo " -- Joshua Randall <jcrandall@alum.mit.edu>  Thu, 4 Feb 2016 11:31:00 -0000" >> debian/changelog
+	echo "" >> debian/changelog
+	cat debian/changelog.old >> debian/changelog
+	rm -f debian/changelog.old
+
+	# install build-deps and build
+        apt-get install -y $(awk 'BEGIN {FS=":"} $1=="Build-Depends" {n=split($2,deps,","); for (i=0; i<n; i++) {split(deps[i],pkg," "); if (pkg[1]!="") {print pkg[1]}}}' debian/control)
```

Comment:
This can't possibly be the best way, can it? OTOH, I'm having trouble coming up with something better. It seems like there must be a tool that's exactly like `apt-get build-deps foo` but looks in a specified `debian/control` file instead of insisting on going through `/etc/apt/sources.list`... but what is it? The closest I've come is `dpkg-checkbuilddeps`, which shows the right information (on stderr) but in an only slightly less machine-unreadable format.

The awk script does seem to miss `libselinux-dev` here:

```
# dpkg-checkbuilddeps debian/control
dpkg-checkbuilddeps: Unmet build dependencies: debhelper (>= 9) dh-autoreconf libselinux-dev
```
```
# awk 'BEGIN {FS=":"} $1=="Build-Depends" {n=split($2,deps,","); for (i=0; i<n; i++) {split(deps[i],pkg," "); if (pkg[1]!="") {print pkg[1]}}}' debian/control
debhelper
dh-autoreconf
```

(I don't know whether that actually affects the resulting package, so I'm assuming it "might".)

This seems to do the parsing and installing properly...but at the cost of a couple hundred megabytes of devscripts dependencies, which seems a bit much:

```
apt-get install devscripts equivs
mk-build-deps -i --tool 'apt-get --no-install-recommends -y'
```

It seems maybe this is Hard because we're expected to either do it by hand, or go all the way, and build in a chroot with pbuilder...?

Or, since the version number is hardcoded, maybe we should just hardcode the dependencies too.